### PR TITLE
Multi User logic

### PR DESCRIPTION
**Pull request by nanstc**
This PR contains: 

- Removal of a final instance of employee in employee repo that gets overwritten when more than 1 user logged in
- Reorganize employee relationship in test-server.ldif so the following is true: 
` user is under approver 1, approver 2 (CC8) 
user2 is somewhere else (CC13)

approver 1, 2 are in same cost centre as user (CC8)
approver 3 is somewhere else (CC7)

admin1 is IT staff (CC20)
admin2 is IT god   (CC20) `

- Added guarding to prevent nullPtrException when user ID does not yield role
- Renamed ` EmployeeAuthenticationProvider ` to ` LoginController `

Test case: if you make a request with user. You can change the request status by approving the request using concurrently logged in approver1 and 2.

### CODE REVIEW

File: src/main/java/com/coastcapitalsavings/auth/WebSecurityConfig.java [hunk 1]
```diff
@@ -67,22 +65,27 @@ protected void configure(HttpSecurity http) throws Exception {
...
-                .antMatchers(HttpMethod.PUT, "/requests").hasAnyAuthority("ROLE_APPROVER")
-                .antMatchers("/**").hasAuthority("ROLE_IT APPROVER")
-                .anyRequest().fullyAuthenticated()
+                    .antMatchers(HttpMethod.GET, "/login").permitAll()
+                    .antMatchers("/scripts/**").permitAll()
+                    .antMatchers("/*.js").permitAll()
+                    .antMatchers("/*.css").permitAll()
+                    .antMatchers(HttpMethod.POST, "/request").permitAll()
+                    .antMatchers(commonEndpoints).
+                        hasAnyAuthority("ROLE_USER", "ROLE_APPROVER", "ROLE_ADMINISTRATOR", "ROLE_IT APPROVER")
```
Comment (csemiao) [inline src/main/java/com/coastcapitalsavings/auth/WebSecurityConfig.java]: Should be ROLE_IT_APPROVER

Comment (csemiao) [inline src/main/java/com/coastcapitalsavings/auth/WebSecurityConfig.java]: Maybe make it ROLE_IT_ADMIN for consistency

Comment (csemiao) [inline src/main/java/com/coastcapitalsavings/auth/WebSecurityConfig.java]: by that I mean ROLE_ADMININSTRATOR could be ROLE_IT_USER or ROLE_IT_ADMIN

Comment (nanstc) [inline src/main/java/com/coastcapitalsavings/auth/WebSecurityConfig.java]: Unfortunately this is a defect in our DB. Making a ticket for polishing maybe after demo.

**Pull request merged**